### PR TITLE
#935 - Check for authentication errors prior to execution.

### LINKED ISSUE #935: Add check for authentication errors before execution
Prior to execution, WPGraphQL should check if there are authentication errors. 

If there are errors, execution should be prevented. 

For example, an expired JWT Token should not execute, but return an auth error and 403 status. Same with an invalid cookie for cookie auth. 

Related to: https://github.com/wp-graphql/wp-graphql-jwt-authentication/issues/38

### PR DESCRIPTION
**Pull request by jasonbahl**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This adds a check for authentication errors prior to execution of the request. 

If there are no authentication errors, execution proceeds as normal. If there are authentication errors, a 403 is returned and execution is prevented.

Does this close any currently open issues?
------------------------------------------
fixes #935 
also related: https://github.com/wp-graphql/wp-graphql-jwt-authentication/issues/38

### CODE REVIEW

File: src/Request.php [hunk 1]
```diff
@@ -112,16 +112,103 @@ private function before_execute() {
...
+	 *
+	 * @access protected
+	 *
+	 * @throws \Exception
+	 */
+	protected function has_authentication_errors() {
+
+		$authentication_errors = false;
+
+		if ( ! empty( $result ) ) {
```
Comment (renatonascalves) [inline src/Request.php]: Where this variable is being set? I'm seeing it.

File: src/Request.php [hunk 2]
```diff
@@ -112,16 +112,103 @@ private function before_execute() {
...
+	/**
+	 * Checks authentication errors.
+	 *
+	 * False will mean there are no detected errors and
+	 * execution will continue.
+	 *
+	 * Anything else (true, WP_Error, thrown exception, etc) will prevent execution of the GraphQL
+	 * request.
+	 *
+	 * @return boolean
```
Comment (renatonascalves) [inline src/Request.php]: `@return` is the last option in the PHPDoc. The order should be `access`, `throws` and `return.`

File: src/Request.php [hunk 3]
```diff
@@ -112,16 +112,103 @@ private function before_execute() {
...
+		return $this->filtered_authentication_errors( $authentication_errors );
+	}
+
+	/**
+	 * Filter Authentication errors. Allows plugins that authenticate to hook in and prevent
+	 * execution if Authentication errors exist.
+	 *
+	 * @param $authentication_errors
+	 * @access protected
+	 *
```
Comment (renatonascalves) [inline src/Request.php]:
```suggestion
```

File: src/Request.php [hunk 4]
```diff
@@ -112,16 +112,103 @@ private function before_execute() {
...
+	 * @return boolean
+	 */
+	protected function filtered_authentication_errors( $authentication_errors = false ) {
+
+		/**
+		 * If false, there are no authentication errors. If true, execution of the
+		 * GraphQL request will be prevented and an error will be thrown.
```
Comment (renatonascalves) [inline src/Request.php]: Missing the params that are being sent...

File: src/Request.php (hunk 1 above)
Comment (jasonbahl) [reply src/Request.php]: oof. I did a refactor before opening the PR and didn't clean up after myself. 🤦‍♂ 

Thanks!

**Comment by jasonbahl** [comment]
closes #935

**Pull request merged**